pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,255 @@
+			reportTemplate: &ReportQueryTemplateContext{
+				Namespace:         testNamespace,
+				Reports:           nil,
+				ReportQuery:       NewReportQueryCustom(ds1, "SELECT labels['namespace'], FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ""},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:36: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource  dependency not found",
+			expectOutput: "",
+		},
+		{
+			name: "valid report query with valid templating but PrestoTable contains empty schema field returns error and expected error output",
+			reportTemplate: &ReportQueryTemplateContext{
+				Namespace:         testNamespace,
+				Reports:           nil,
+				ReportQuery:       NewReportQueryCustom(ds1, "SELECT labels['namespace'] FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom(ds1.Status.TableRef.Name, testNamespace, "", nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:35: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: prestoTable %s either contains an empty field in the Catalog, Schema, and/or TableName Status structure", ds1.Status.TableRef.Name),
+			expectOutput: "",
+		},
+	}
+
+	for _, testCase := range testTable {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			output, err := RenderQuery(testCase.reportTemplate, testCase.templateContext)
+
+			if testCase.expectErr {
+				assert.EqualErrorf(t, err, testCase.expectErrMsg, "expected that RenderQuery would return the correct error message")
+			} else {
+				assert.NoErrorf(t, err, "expected the report would return no error, but got an error.")
+				assert.EqualValuesf(t, testCase.expectOutput, output, "expected that RenderQuery would return the expected output")
+			}
+		})
+	}
+}
+
+// NewPrestoTableCustom allows us to insert a custom name, as opposed to testhelpers.NewPrestoTable
+func NewPrestoTableCustom(name, namespace, schema string, columns []presto.Column) *v1alpha1.PrestoTable {
```

Comment:
Yeah let's just put this into testhelpers, remove the `Custom` from the name

Reply:
I did that initially and there was an import cycle w/ the reporting package and testhelpers

Reply:
Ah. Yeah, our packages need to be reorganized. Well for now you can keep it here, but don't export it, and perhaps put the word test into the function name., eg: `newTestPrestoTable`

Reply:
Updated - renamed to include test and unexported the functions.